src/main/scala/rocket/AMOALU.scala

```diff
@@ -3,104 +3,109 @@
 
 package freechips.rocketchip.rocket
 
-import Chisel._
+import chisel3._
+import chisel3.util._
 import freechips.rocketchip.config.Parameters
 
 class StoreGen(typ: UInt, addr: UInt, dat: UInt, maxSize: Int) {
-  val size = typ(log2Up(log2Up(maxSize)+1)-1,0)
-  def misaligned =
-    (addr & ((UInt(1) << size) - UInt(1))(log2Up(maxSize)-1,0)).orR
+  val size: UInt = typ(log2Up(log2Up(maxSize) + 1) - 1, 0)
 
-  def mask = {
-    var res = UInt(1)
+  def misaligned: Bool =
+    (addr & ((1.U << size).asUInt - 1.U) (log2Up(maxSize) - 1, 0)).orR
```

Comment:
no need `asUInt` here, if you wanna make IDEA happy, you just need use `: UInt` to handle the type hint.